purchase_discount/models/product_supplierinfo.py

```diff
@@ -27,9 +27,19 @@ def _get_po_to_supplierinfo_synced_fields(self):
     def create(self, vals_list):
         """Insert discount (or others) from context from purchase.order's
         _add_supplier_to_product method"""
-        for vals in vals_list:
-            product_tmpl_id = vals.get("product_tmpl_id")
-            po_line_map = self.env.context.get("po_line_map", {})
+        po_line_map = self.env.context.get("po_line_map")
+        if not po_line_map:
+            return super().create(vals_list)
+        for vals in filter(
+            lambda x: x.get("product_tmpl_id", x.get("product_id")), vals_list
```

Comment:
This filter seems now incorrect if we fallback to the other value.

Reply:
Not important now, but it just meant that we could go with that dictionary as we could get the template from the product_id